services/QuillLMS/app/models/concerns/flagset.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# This concern is only intended to be included in the User model
+module Flagset
+  #FLAGS = %w(production archived alpha beta gamma private)
+  extend ActiveSupport::Concern
+
+
+  included do
+    FLAGSETS = {
+
+      production: {
+        display_name: 'production',
+        flags: {
+          'production' =>        { display_name: 'Production' }
+        }
+      },
+
+      alpha: {
+        display_name: 'Alpha',
+        flags: {
+          'alpha' =>             { display_name: 'Alpha' },
+          'evidence_beta1' =>   { display_name: 'Evidence Beta 1' },
+          'evidence_beta2' =>   { display_name: 'Evidence Beta 2' },
+          'beta' =>              { display_name: 'Beta' },
+          'college_board' =>     { display_name: 'College Board' },
+          'production' =>        { display_name: 'Production' }
+        }
+      },
+
+      evidence_beta1: {
+        display_name: 'Evidence Beta 1',
+        flags: {
+          'evidence_beta1' =>   { display_name: 'Evidence Beta 1' },
+          'evidence_beta2' =>   { display_name: 'Evidence Beta 2' },
+          'production' =>        { display_name: 'Production' }
+        }
+      },
+
+      evidence_beta2: {
+        display_name: 'Evidence Beta 2',
+        flags: {
+          'evidence_beta1' =>   { display_name: 'Evidence Beta 1' },
+          'production' =>        { display_name: 'Production' }
+        }
+      },
+
+      beta: {
+        display_name: 'Beta',
+        flags: {
+          'beta' =>              { display_name: 'Beta' },
+          'production' =>        { display_name: 'Production' }
+        }
+      },
+
+      college_board: {
+        display_name: 'College Board',
+        flags: {
+          'college_board' =>     { display_name: 'College Board' },
+          'production' =>        { display_name: 'Production' }
+        }
+      }
+    }
+
+    validates :flagset, inclusion: { in: FLAGSETS.keys.map(&:to_s) }
+  end
+
+  def activity_viewable?(activity)
+    return false unless activity.is_a?(::Activity)
+
+    user_flagset = FLAGSETS.fetch(flagset.to_sym){ return false }
```

Comment:
Nice use of a return block with `fetch`

The variable `user_flagset` indicates a coupling to the user model.  If the intent is only to use this as a user concern, either rename I would rename it `UserFlagset`.  If this model is meant to be included elsewhere, I would rename the `user_flagset` variable to something more generic.

Reply:
> On a related note is `user_flagset` necessary at all since there is `validates :flagset, inclusion: { in: FLAGSETS.keys.map(&:to_s) }`. In other words shouldn't the fetch always succeed?

The flagsets will change in the future, I prefer to keep the guard here.